### PR TITLE
Add johnnydep, wimpy

### PR DESCRIPTION
**Pull request by ickc**
@conda-forge/help-python

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by ickc** [comment]
@wimglenn, I'm trying to add your package to conda-forge, and everything is passing except the Windows platform. I try to run `johnnydep johnnydep` as a test, but it fails on Windows. How should that command be run on Windows? Thanks.

File: recipes/johnnydep/LICENSE [hunk 1]
```diff
@@ -0,0 +1,21 @@
+MIT License
```
Comment (chrisburr) [inline recipes/johnnydep/LICENSE:1]: This file isn't needed as it's included in the source tarball and the `licence_file` field looks there automatically. It's also preferable as it ensures the file stays up to date if the licence ever changes.

File: recipes/wimpy/LICENSE [hunk 2]
```diff
@@ -0,0 +1,21 @@
+MIT License
```
Comment (chrisburr) [inline recipes/wimpy/LICENSE:1]: This file isn't needed as it's included in the source tarball and the `licence_file` field looks there automatically. It's also preferable as it ensures the file stays up to date if the licence ever changes.

File: recipes/johnnydep/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+test:
+  imports:
+    - johnnydep
+  commands:
+    - johnnydep johnnydep
+
+about:
+  home: https://github.com/wimglenn/johnnydep
+  license: MIT
+  license_family: MIT
```
Comment (chrisburr) [inline recipes/johnnydep/meta.yaml:52]: There is no need for this field any more as it's computed automatically:

```suggestion
```

File: recipes/wimpy/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+test:
+  imports:
+    - wimpy
+
+about:
+  home: https://github.com/wimglenn/wimpy
+  license: MIT
+  license_family: MIT
+  license_file: LICENSE
```
Comment (chrisburr) [inline recipes/wimpy/meta.yaml:34]: There is no need for this field any more as it's computed automatically:

```suggestion
```

**Comment by wimglenn** [comment]
@ickc  It works fine on Windows, but their CI is failing to print the output 

```
  File "C:\Miniconda\conda-bld\johnnydep_1607733309412\_test_env\lib\encodings\cp1252.py", line 19, in encode
    return codecs.charmap_encode(input,self.errors,encoding_table)[0]
UnicodeEncodeError: 'charmap' codec can't encode characters in position 211-213: character maps to <undefined>
```

this error occurs right when johnnydep would have begun to print non-ascii characters like └──

maybe you should just change your test to something simpler like `johnnydep --version` or `johnnydep --help`.

File: recipes/johnnydep/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+    - packaging >=17
+    - wheel >=0.32.0
+    - setuptools >=38.3
+    - pkginfo >=1.4.2
+
+test:
+  imports:
+    - johnnydep
+  commands:
+    - johnnydep johnnydep
```
Comment (wimglenn) [inline recipes/johnnydep/meta.yaml:47]:
```suggestion
    - johnnydep --version
```

File: recipes/wimpy/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+{% set version = "0.6" %}
+{% set sha256 = "014d21a31980eab48b92c4210e9fdba643b79f7081c64ea9bc97f2be9db9bd58" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/{{ gh_author }}/{{ name }}/archive/v{{ version }}.tar.gz
+  sha256: {{ sha256 }}
```
Comment (chrisburr) [inline recipes/wimpy/meta.yaml:12]: Avoiding variables which are only used once makes it easier for the bot to make updates:

```suggestion
  url: https://github.com/wimglenn/{{ name }}/archive/v{{ version }}.tar.gz
  sha256: 014d21a31980eab48b92c4210e9fdba643b79f7081c64ea9bc97f2be9db9bd58
```

(it should handle this scenario regardless but as I had other comments I might as well include this one)

File: recipes/wimpy/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,40 @@
+{% set name = "wimpy" %}
+{% set gh_author = "wimglenn" %}
+{% set version = "0.6" %}
+{% set sha256 = "014d21a31980eab48b92c4210e9fdba643b79f7081c64ea9bc97f2be9db9bd58" %}
```
Comment (chrisburr) [inline recipes/wimpy/meta.yaml:4]: See other comment:

```suggestion
{% set version = "0.6" %}
```

File: recipes/johnnydep/meta.yaml [hunk 8]
```diff
@@ -0,0 +1,59 @@
+{% set name = "johnnydep" %}
+{% set gh_author = "wimglenn" %}
+{% set version = "1.7" %}
+{% set sha256 = "1e7bd54f09437a929bd29cd3d229b9579423affd5348ce1ec995faf5958440da" %}
```
Comment (chrisburr) [inline recipes/johnnydep/meta.yaml:4]: See other comment:
```suggestion
{% set version = "1.7" %}
```

File: recipes/johnnydep/meta.yaml [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/{{ gh_author }}/{{ name }}/archive/v{{ version }}.tar.gz
+  sha256: {{ sha256 }}
```
Comment (chrisburr) [inline recipes/johnnydep/meta.yaml:12]: See other comment:
```suggestion
  url: https://github.com/wimglenn/{{ name }}/archive/v{{ version }}.tar.gz
  sha256: 1e7bd54f09437a929bd29cd3d229b9579423affd5348ce1ec995faf5958440da
```

**Review by chrisburr — COMMENTED** [review]
The Windows failure can be ignored as this is a `noarch: python` package and it seems like an upstream bug which can be reported with this traceback:
```log
Traceback (most recent call last):
  File "C:\Miniconda\conda-bld\johnnydep_1607733309412\_test_env\Scripts\johnnydep-script.py", line 9, in <module>
    sys.exit(main())
  File "C:\Miniconda\conda-bld\johnnydep_1607733309412\_test_env\lib\site-packages\johnnydep\cli.py", line 65, in main
    print(dist.serialise(fields=args.fields, format=args.output_format, recurse=args.recurse))
  File "C:\Miniconda\conda-bld\johnnydep_1607733309412\_test_env\lib\site-packages\colorama\ansitowin32.py", line 41, in write
    self.__convertor.write(text)
  File "C:\Miniconda\conda-bld\johnnydep_1607733309412\_test_env\lib\site-packages\colorama\ansitowin32.py", line 162, in write
    self.write_and_convert(text)
  File "C:\Miniconda\conda-bld\johnnydep_1607733309412\_test_env\lib\site-packages\colorama\ansitowin32.py", line 190, in write_and_convert
    self.write_plain_text(text, cursor, len(text))
  File "C:\Miniconda\conda-bld\johnnydep_1607733309412\_test_env\lib\site-packages\colorama\ansitowin32.py", line 195, in write_plain_text
    self.wrapped.write(text[start:end])
  File "C:\Miniconda\conda-bld\johnnydep_1607733309412\_test_env\lib\site-packages\colorama\ansitowin32.py", line 41, in write
    self.__convertor.write(text)
  File "C:\Miniconda\conda-bld\johnnydep_1607733309412\_test_env\lib\site-packages\colorama\ansitowin32.py", line 162, in write
    self.write_and_convert(text)
  File "C:\Miniconda\conda-bld\johnnydep_1607733309412\_test_env\lib\site-packages\colorama\ansitowin32.py", line 190, in write_and_convert
    self.write_plain_text(text, cursor, len(text))
  File "C:\Miniconda\conda-bld\johnnydep_1607733309412\_test_env\lib\site-packages\colorama\ansitowin32.py", line 195, in write_plain_text
    self.wrapped.write(text[start:end])
  File "C:\Miniconda\conda-bld\johnnydep_1607733309412\_test_env\lib\encodings\cp1252.py", line 19, in encode
    return codecs.charmap_encode(input,self.errors,encoding_table)[0]
UnicodeEncodeError: 'charmap' codec can't encode characters in position 211-213: character maps to <undefined>
```

File: recipes/wimpy/meta.yaml (hunk 4 above)
Comment (ickc) [reply recipes/wimpy/meta.yaml]: After I removed the line `license_file: LICENSE` for both of them the bot is now complaining. Is it ok?

**Comment by ickc** [comment]
Thanks all. Apparently I could have accepted the suggested change here, but I did it in git and pushed them. I add all your suggested changes.

I just worry that the bot now complain `license_file entry is missing, but is required.` for both packages. Would that be ok, @chrisburr?

**Comment by chrisburr** [comment]
You still need the field in the YAML, you just don't need to include the file itself

**Pull request merged**